### PR TITLE
worked on creating payment intent for a customer

### PR DESCRIPTION
**Pull request by MUGABA**
### What this PR does (required):
- This pr creates a payment intent for a customer

Screenshot 2022-02-02 152427

### CODE REVIEW

File: client/src/components/PaymentMethods/paymentMethods.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const PaymentMethods = (): JSX.Element => {
+  const createPaymentProfile = async () => {
+    console.log('Am called');
+
+    const result = await PaymentMethod();
+
+    console.log(result);
+    // console.log(result);
+
+    // <Redirect to={`${result.url}`} />;
```
Comment (ezeilo-su) [inline client/src/components/PaymentMethods/paymentMethods.tsx:16]: Please remove comment

File: client/src/components/PaymentMethods/paymentMethods.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import PaymentMethod from '../../helpers/APICalls/paymentMethods';
+import SettingHeader from '../SettingsHeader/SettingsHeader';
+
+const PaymentMethods = (): JSX.Element => {
+  const createPaymentProfile = async () => {
+    console.log('Am called');
+
+    const result = await PaymentMethod();
+
+    console.log(result);
```
Comment (ezeilo-su) [inline client/src/components/PaymentMethods/paymentMethods.tsx:13]: Please watch for unnecessary logs to the console. Going forward, we should be cleaning up our logs done during development. Thanks!

File: client/src/components/PaymentMethods/paymentMethods.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { Redirect } from 'react-router-dom';
+import PaymentMethod from '../../helpers/APICalls/paymentMethods';
+import SettingHeader from '../SettingsHeader/SettingsHeader';
+
+const PaymentMethods = (): JSX.Element => {
+  const createPaymentProfile = async () => {
+    console.log('Am called');
```
Comment (ezeilo-su) [inline client/src/components/PaymentMethods/paymentMethods.tsx:9]: Remove this one

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Hey @MUGABA, good work with this one. I have left you a few comments to address.
Happy coding!

File: server/controllers/paymentMethods.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  res.send({ url });
+
+  try {
+    const setupIntent = await stripe.setupIntents.create({
+      customer: userProfle.stripeAccountId,
+      payment_method_types: ["card"],
+    });
+  } catch (error) {
+    res.status(400);
+    throw new Error("Payment Intent not created");
```
Comment (moffatethan) [inline server/controllers/paymentMethods.js:40]:
```suggestion
    throw new Error("SetupIntent not created");
```

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — APPROVED** [review]

**Pull request merged**